util/src/main/java/io/camunda/zeebe/util/FileUtil.java

```diff
@@ -34,10 +35,9 @@ private FileUtil() {}
 
   /**
    * Ensures updates to the given path are flushed to disk, with the same guarantees as {@link
-   * FileChannel#force(boolean)}. This can be used for both directory and files. Note that if you
-   * already have a file channel open, then use the {@link FileChannel#force(boolean)} method
-   * directly. This method is mostly useful when flushing directories after file creation and/or
-   * file rename operations.
+   * FileChannel#force(boolean)}. This can be used for files only; for directories, call {@link
+   * #flushDirectory(Path)}. Note that if you already have a file channel open, then use the {@link
+   * FileChannel#force(boolean)} method directly.
```

Comment:
The chosen solution forces the developer to consider whether the file is a directory or not and use the correct method accordingly. I feel like this could be made more clear by changing the method name to `flushFile` or by making flush take a `File` as parameter. Another solution altogether would be to check it inside the flush implementation, making it completely transparent to the caller.

🔧 Please improve the understandability for callers of the flush method.

Reply:
I knew I would get a comment here :smile: 

I opted for this separation instead of a check in `flush` because I wanted it to be more explicit - if you call `flush` directly, then there is no no-op behavior and it should explode. If you want to flush a directory, then you will have the no-op behavior. I wanted it to be explicit because the no-op is only fine for directory.

At the same time, I wouldn't rename `flush` to `flushFile`, since it's actually used in `flushDirectory`. The alternative I see is having a check in `flush`, and if it's a directory, then we opt for the no-op behavior. Passing a `File` wouldn't help since directories are also `File` objects.

So my point is, we have a slight variant when the input is a directory, and it's only in very specific cases that we want to flush directories, all of which we know in advance that it's a directory we're flushing. If you use the other method, because you might be flushing one or the other, then there's a good chance that you don't want to no-op behavior (at least there's no use case at the moment for this). Does that make sense?

I'm open to merging both, but my worry is that the no-op because more unexpected.

Reply:
>only in very specific cases that we want to flush directories

You make a good point here. I guess in most cases, the caller isn't concerned with this and `flush` is fine. In those specific cases we want to flush directories, we just need to remember about the other method. To be fair, intelli-sense already helps with this everytime you type `FileUtil.flush`. I'm okay with leaving it 👍

Reply:
Let's go with this, and if it happens that someone makes the mistake in the future, we can refactor it. My thought was that the more likely mistake is expecting flush to always happen and being surprised that it's a no-op, but...yeah.